### ISSUE TITLE
dmypy does not catch `--warn-unused-ignores` in successive runs

### ISSUE THREAD

**Issue by clambert-jt** [python/mypy#19363]
**Bug Report**

dmypy does not emit `--warn-unused-ignores` warnings after a file changes

**To Reproduce**

```bash
$ dmypy stop
No status file found
$ echo 'print(1) # type: ignore' > a.py
$ dmypy run -- a.py --warn-unused-ignores
Daemon started
a.py:1: error: Unused "type: ignore" comment  [unused-ignore]
Found 1 error in 1 file (checked 1 source file)
$ echo 'print(2) # type: ignore' >> a.py
$ dmypy run -- a.py --warn-unused-ignores
Success: no issues found in 1 source file
```

**Expected Behavior**

The second run of dmypy should at give the first warning as well as an additional warning for the second.

**Actual Behavior**

dmypy does not warn in the second run

**Your Environment**

- Mypy version used: 1.16.1
- Mypy command-line flags: above
- Mypy configuration options from `mypy.ini` (and other config files): n/a
- Python version used: 3.12

**Comment by hauntsaninja** [comment]
I believe this is fixed on master

**Comment by sterliakov** [comment]
Confirmed, works on master.
